server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3483,6 +3483,8 @@ private Ternary<List<Long>, Integer, String[]> searchForDiskOfferingsIdsAndCount
                     diskOfferingSearch.and("displayOffering", diskOfferingSearch.entity().getDisplayOffering(), Op.EQ);
                 }
 
+                diskOfferingSearch.and("activeState", diskOfferingSearch.entity().getState(), Op.EQ);
+
                 SearchCriteria<DiskOfferingVO> sc = diskOfferingSearch.create();
                 sc.setParameters("computeOnly", false);
                 sc.setParameters("activeState", DiskOffering.State.Active);
```

Comment:
```suggestion
                if (state != null) {
                    sc.setParameters("state", state);
                }
```

Reply:
I guess the ideal implementation would be

```
-        if (state != null) {
-            diskOfferingSearch.and("state", diskOfferingSearch.entity().getState(), Op.EQ);
-        }
+        diskOfferingSearch.and("state", diskOfferingSearch.entity().getState(), Op.EQ);
```
...
```
-                sc.setParameters("activeState", DiskOffering.State.Active);
+                if (state != null) {
+                    sc.setParameters("state", state);
+                } else {
+                    sc.setParameters("state", DiskOffering.State.Active);
+                }

```

?

Reply:
cc @Pearl1594 @vishesh92

Reply:
lgtm